### ISSUE TITLE
Can CCLoader  run on esp32?

### ISSUE THREAD

**Issue by Tolomaj** [RedBearLab/CCLoader#37]
Is CCloader runing properly on esp32?

Sketch uploads to esp32 jist fine.

But can here be any problem with this setup ?

**Comment by Tolomaj** [comment]
For others i tried and yes and it works on esp32

**Comment by EntryGoon** [comment]
I'm trying using ESP32 instead of ESP8266, for flashing CC2530 with CCloader. can you please let me know on which pins of the ESP32 I should connect ? it should be GPIO4, GPIO5 and GPIO12 for ESP8266, but I assume used GPIO are not the same for ESP32...

**Comment by Tolomaj** [comment]
Sorry.
It was few months since I played with Ccloader, and I don't remember anymore witch pins that was. 
But after quick look in fork of original repo for esp8266([link to repo](https://github.com/jeanniquini/CCLoader/tree/master)) i find this.
![image](https://github.com/RedBearLab/CCLoader/assets/59420562/0af97630-2c22-4bbb-9a87-21791b4d158e)
i assume you can set your pins in .ino file.

**Comment by EntryGoon** [comment]
Yep i saw that and I have to try this. Thanks a lot for your help :-)

**Comment by ds2k5** [comment]
@EntryGoon 

Hi could you be so kind and help me out....
Did not get FW flashed to CC2530...

**Comment by Tolomaj** [comment]
As i write above, I dont remember exactly how flash the chip. But..

In my case i flash cc254x chip (actually HM-10 but chip should be same).
Because the datasheet is same for CC254x and CC2530 i think flashing should work the same.

I asked Google and found some thinks i remeber on site: https://www.hangar42.nl/ccloader.
Like pins used for flashing. (I know this is arduino but on CC side it doesn't matter).
![image](https://github.com/RedBearLab/CCLoader/assets/59420562/6536abc7-56ba-4b15-88cd-78097eccc2aa)
And I assume the esp32 pins is this:
> ![image](https://user-images.githubusercontent.com/59420562/275079382-0af97630-2c22-4bbb-9a87-21791b4d158e.png) 

Also, I remember having some problem with BT (probably some my fault).
But when i try flash some blink code it works.

Unfortunately I can't help more than that.

**Comment by ds2k5** [comment]
@Tolomaj   thanks

had no luck with the ESP32

my colleage rent my his: Chipcon AS SmartRF04DD 
so I could flash the hex file with the TI-Software on a Windows Box

**Comment by JonRobert** [comment]
I've had good luck using a NodeMCU v2 using pins:
// define pins for NodeMCU
   int DD    = 14;  // data
   int DC    = 4;   // Clock
   int RESET = 5;
   int led   = 2

However now I'm trying to use a SEEED XIAO ESP32 C3 (because they are small and I have a couple of extra ones.
I'm using Pins:
// define pins for XIAO esp32-c3
   int RESET = 7;    //D5;    // i2c SCL
   int DC    = 6;    //D4;    // i2c SDA
   int DD    = 8;    //D8;    // SPI SCLK
   // int LED   = ;

The results seems to "hang" after (or trying to ) write the 1st block.
Any thoughts' 

```

Waiting for respond from Arduino...

Uploading firmware...

1

```

**Comment by hacst** [comment]
Can verify the hang with a ESP32 C3 SuperMini. Worked without any problems on a ESP32 S3 DevKit though.

**Comment by JonRobert** [comment]
Yes, but it will take some time, I’m in between mother boards.

 

John
